ChatRPG/Services/EmailSender.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Drawing.Printing;
+using Microsoft.AspNetCore.Identity.UI.Services;
+using MailKit.Net.Smtp;
+using MimeKit;
+using MimeKit.Text;
+
+namespace ChatRPG.Services;
+
+public class EmailSender : IEmailSender
+{
+
+    private readonly string _senderEmail;
+    private readonly string _senderPassword;
+
+    public EmailSender(IConfiguration configuration)
+    {
+        _senderEmail = configuration.GetSection("ChatRPGEmail").GetValue<string>("Email");
```

Comment:
To ting:

1. Hvis vi _altid_ forventer at skulle sende emails, så kan du bruge `.GetRequiredValue<string>("Email")` og håndtere exception på passende vis.
2. Hvis vi _ikke altid_ forventer at sende mails (fx. i tests eller lokale miljøer) så bør der måske være en option der "disabler" den her klasse på en eller anden måde.

Reply:
Altså for at man skal kunne reset sit password osv så skal der altid sendes en mail. Er det det du mener med nr 1?
Muligvis kunne man gøre sådan at hvis der ikke er defined en email i configuration, så sendes der ikke en mail.

Reply:
ift til at disable klassen kan det måske godt blive lidt funky. Men man kunne bare have en boolean value gemt i configuration der bestemmer om man skal sende mails

Reply:
Yeah, jeg tænker bare når man tester lokalt behøver den ikke spinde en email-server op eller noget i den dur. Så det vil være fint med en boolean den lige kan tjekke inden den laver en masse gøjl i den her klasse

Reply:
> 2. Hvis vi _ikke altid_ forventer at sende mails (fx. i tests eller lokale miljøer) så bør der måske være en option der "disabler" den her klasse på en eller anden måde.

f.eks en mocked run configuration

Reply:
Har gjort sådan at man skal tilføje en boolean sammen med email credentials i user secrets før at den sender mails. Hvis boolean ikke er inkluderet, så er default value false, og derfor behøver man så heller ikke at have email credentials defined. Har implementeret tjekket inden at SendEmailAsync bliver kaldt. Det letteste ville have været at implementere tjekket inde i funktionen, men så strider det bare imod navnet på funktionen, idet det implyer at vi ved at vi skal sende en mail.

Reply:
Og ift `.GetRequiredValue<string>("Email")` så er jeg ikke sikker på om den rent faktisk findes, måske er jeg bare stupid idk.

Reply:
looks good to me